lte/gateway/c/core/oai/tasks/grpc_service/AmfServiceImpl.cpp

```diff
@@ -119,14 +118,22 @@ Status AmfServiceImpl::SetSmfSessionContext(
   itti_msg.m5gsm_cause = (m5g_sm_cause_t)req_m5g.m5gsm_cause();
 
   itti_msg.session_ambr.uplink_unit_type = req_m5g.subscribed_qos().br_unit();
-  itti_msg.session_ambr.uplink_units = req_m5g.subscribed_qos().apn_ambr_ul();
-
   itti_msg.session_ambr.downlink_unit_type = req_m5g.subscribed_qos().br_unit();
-  itti_msg.session_ambr.downlink_units = req_m5g.subscribed_qos().apn_ambr_dl();
 
-  // authorized qos profile
-  itti_msg.qos_list.qos_flow_req_item.qos_flow_identifier =
-      req_m5g.subscribed_qos().qos_class_id();
+  if (!req_m5g.qos().max_req_bw_ul() && !req_m5g.qos().max_req_bw_dl()) {
+    // APN ambr value if policy is not attached
+    itti_msg.session_ambr.uplink_units = req_m5g.subscribed_qos().apn_ambr_ul();
+    itti_msg.session_ambr.downlink_units =
+        req_m5g.subscribed_qos().apn_ambr_dl();
+    itti_msg.qos_list.qos_flow_req_item.qos_flow_identifier =
+        req_m5g.subscribed_qos().qos_class_id();
+  } else {
+    // Policy ambr value if policy attached
+    itti_msg.session_ambr.uplink_units = req_m5g.qos().max_req_bw_ul();
+    itti_msg.session_ambr.downlink_units = req_m5g.qos().max_req_bw_dl();
+    itti_msg.qos_list.qos_flow_req_item.qos_flow_identifier =
+        req_m5g.qos().qci();
```

Comment:
can you add more details on how is this qos policy configured?

Reply:
It is configured through the GUI, We should have an active policy.
Attaching a screen shot where policy is not attached:
![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/94469973/159199041-dd2f1316-2a58-4c37-b7b6-a4a32fbde66a.png)